### PR TITLE
oatpp-sqlite: update sqlite3, use transitive_headers, use rm_safe

### PR DESCRIPTION
**Pull request by toge**
Specify library name and version:  **oatpp-sqlite/***

---

- [x] I've read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md).
- [x] I've used a [recent](https://github.com/conan-io/conan/releases/latest) Conan client version close to the [currently deployed](https://github.com/conan-io/conan-center-index/blob/master/.c3i/config_v1.yml#L6).
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by toge** [comment]
Missing dependencies is again.
```
ERROR: Missing prebuilt package for 'oatpp/1.3.0'
```

**Comment by RubenRBS** [comment]
Looks like an epoch issue. Let me rebuild this internally to bump it, see if it correctly fetches msvc 193 binaries after that :) (Feel free to ping me next time if I miss any of these in your PRs!)

**Comment by RubenRBS** [comment]
Rebuilding the missing binary now

**Comment by RubenRBS** [comment]
Oops, just realized how the dependencies on oatpp work, building the remaining binaries :)

File: recipes/oatpp-sqlite/all/conanfile.py [hunk 1]
```diff
@@ -34,17 +34,14 @@ def config_options(self):
...
+            self.options.rm_safe("fPIC")
 
     def layout(self):
         cmake_layout(self, src_folder="src")
 
     def requirements(self):
-        self.requires(f"oatpp/{self.version}")
-        self.requires("sqlite3/3.39.4")
+        self.requires(f"oatpp/{self.version}", transitive_headers=True)
+        self.requires("sqlite3/3.43.1")
```
Comment (mayeut) [inline recipes/oatpp-sqlite/all/conanfile.py:44]:
```suggestion
        self.requires("sqlite3/3.44.0")
```

**Review by ericLemanissier — APPROVED** [review]

**Review by RubenRBS — APPROVED** [review]
Thanks (Sorry for the delay!)

**Pull request merged**